x-pack/plugins/ml/server/shared_services/providers/modules.ts

```diff
@@ -4,86 +4,59 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { APICaller, SavedObjectsClientContract } from 'kibana/server';
-import { LicenseCheck } from '../license_checks';
-import { DataRecognizer, RecognizeResult } from '../../models/data_recognizer';
-import {
-  Module,
-  DatafeedOverride,
-  JobOverride,
-  DataRecognizerConfigResponse,
-} from '../../../common/types/modules';
+import { APICaller, KibanaRequest, SavedObjectsClientContract } from 'kibana/server';
+import { DataRecognizer } from '../../models/data_recognizer';
+import { SharedServicesChecks } from '../shared_services';
 
 export interface ModulesProvider {
   modulesProvider(
     callAsCurrentUser: APICaller,
+    request: KibanaRequest,
     savedObjectsClient: SavedObjectsClientContract
   ): {
-    recognize(indexPatternTitle: string): Promise<RecognizeResult[]>;
-    getModule(moduleId?: string): Promise<Module | Module[]>;
-    saveModuleItems(
-      moduleId: string,
-      prefix: string,
-      groups: string[],
-      indexPatternName: string,
-      query: any,
-      useDedicatedIndex: boolean,
-      startDatafeed: boolean,
-      start: number,
-      end: number,
-      jobOverrides: JobOverride[],
-      datafeedOverrides: DatafeedOverride[],
-      estimateModelMemory?: boolean
-    ): Promise<DataRecognizerConfigResponse>;
+    recognize: DataRecognizer['findMatches'];
+    getModule: DataRecognizer['getModule'];
+    listModules: DataRecognizer['listModules'];
+    setupModuleItems: DataRecognizer['setupModuleItems'];
   };
 }
 
-export function getModulesProvider(isFullLicense: LicenseCheck): ModulesProvider {
+export function getModulesProvider({
+  isFullLicense,
+  getHasMlCapabilities,
+}: SharedServicesChecks): ModulesProvider {
   return {
-    modulesProvider(callAsCurrentUser: APICaller, savedObjectsClient: SavedObjectsClientContract) {
-      isFullLicense();
+    modulesProvider(
+      callAsCurrentUser: APICaller,
+      request: KibanaRequest,
+      savedObjectsClient: SavedObjectsClientContract
+    ) {
+      const hasMlCapabilities = getHasMlCapabilities(request);
+      const dr = dataRecognizerFactory(callAsCurrentUser, savedObjectsClient);
       return {
-        recognize(indexPatternTitle: string) {
-          const dr = dataRecognizerFactory(callAsCurrentUser, savedObjectsClient);
-          return dr.findMatches(indexPatternTitle);
+        async recognize(...args) {
+          isFullLicense();
+          await hasMlCapabilities(['canCreateJob']);
```

Comment:
Wondering why the user needs to have `canCreateJob`? Is it required by what `recognize` is doing, or is it because 'recognize' normally leads on to a user creating jobs from a module that matches their data?

Reply:
rather than `canCreateJob ` we could use `canAccessML` or `canGetJobs`, but both are non-admin permissions.
It depends on whether we want the "recognizing" process to be an admin only feature?

Reply:
I think we should keep with `canCreateJob` for now then. It is consistent with the check on the `recognize` endpoint, and this call is about seeing which jobs can be created for the index pattern that is supplied.